src/contract/ens/public_resolver.rs

```diff
@@ -0,0 +1,258 @@
+use crate::{
+    api::Eth,
+    contract::{Contract, Options},
+    signing::NameHash,
+    types::{Address, Bytes, TransactionId, U256},
+    Transport,
+};
+
+type ContractError = crate::contract::Error;
+
+// See https://github.com/ensdomains/resolvers/blob/master/contracts/Resolver.sol for resolver interface.
+#[derive(Debug, Clone)]
+pub struct PublicResolver<T: Transport> {
+    contract: Contract<T>,
+}
+
+impl<T: Transport> PublicResolver<T> {
+    pub fn new(eth: Eth<T>, resolver_addr: Address) -> Self {
+        // See https://github.com/ensdomains/ens-contracts for up to date contracts.
+        let bytes = include_bytes!("PublicResolver.json");
+
+        let contract = Contract::from_json(eth, resolver_addr, bytes).expect("Contract Creation");
+
+        Self { contract }
+    }
+}
+
+impl<T: Transport> PublicResolver<T> {
+    // https://docs.ens.domains/contract-api-reference/publicresolver#get-contract-abi
```

Comment:
```suggestion
    /// https://docs.ens.domains/contract-api-reference/publicresolver#get-contract-abi
```

Please use rust doc comment format everywhere. Ideally we should have a one sentence description, followed by an empty line before the link as well.